### ISSUE TITLE
can init be combined with binseg?

### ISSUE THREAD

**Issue by tdhock** [diego-urgell/BinSeg#19]
hi @diego-urgell looking at RcppInterface.cpp I see
```cpp
    algo -> init(&data[0], data.size(), numCpts, dist, minSegLen, &params_mat[0]);
    algo -> binseg();
```
Is there any reason you would run init without binseg (or vice versa) ? If so can you maybe add a comment to explain/clarify? If not can you please combine them to avoid confusion?

**Comment by diego-urgell** [comment]
I completely agree with you. I think it would be possible to move the `init` call into the `binseg` method.
